tester.go

```diff
@@ -26,12 +29,24 @@ type T struct {
 	SegmentWriteKey     string
 	WebhookBucket       string
 	WebhookAuthUsername string
+	ReportFileName      string
```

Comment:
Slight nit, but from a DI perspective, I think it would be better for this component to not know about files at all, and instead accept a writer. The main command should pass it a fileWriter or a Discard writer as needed.